mypy/types.py

```diff
@@ -833,6 +834,8 @@ def __init__(self, items: List[Type], fallback: Instance, line: int = -1,
                  column: int = -1, implicit: bool = False) -> None:
         self.items = items
         self.fallback = fallback
+        # TODO: assert not (isinstance(fallback, Instance) and fallback.type and
```

Comment:
And again, not sure why assert fails.  I'll commit fixes to the other problems first.